src/minigames/scrubMinigame.ts

```diff
@@ -0,0 +1,78 @@
+import { Application, Graphics, Point } from "pixi.js";
+import { Minigame } from "./minigame";
+
+interface DirtySpotDelegate {
+  markCleaned(spot: DirtySpot): void;
+}
+
+class DirtySpot {
+  private health = 100;
+  private pointerDown = false;
+
+  constructor(
+    private readonly app: Application,
+    private readonly delegate: DirtySpotDelegate,
+  ) {}
+
+  getObject() {
+    const spot = new Graphics();
+
+    const x = (this.app.screen.width - 100) * Math.random() + 50;
+    const y = (this.app.screen.height - 100) * Math.random() + 50;
+    spot.rect(-50, -50, 100, 100);
+    spot.fill(0xabcdef);
+    spot.position = new Point(x, y);
+
+    spot.eventMode = "static";
+    spot.on("mousemove", (event) => {
+      if (!this.pointerDown) return;
+
+      this.health = Math.max(this.health - event.movement.magnitude() * 0.1, 0);
+      spot.alpha = this.health / 100;
+      if (this.health === 0) {
+        this.delegate.markCleaned(this);
+        spot.removeFromParent();
+      }
+    });
+
+    return spot;
+  }
+
+  updatePointerDown(isDown: boolean) {
+    this.pointerDown = isDown;
+  }
+}
+
+export class ScrubMinigame extends Minigame implements DirtySpotDelegate {
+  private remainingDirtySpots: Set<DirtySpot> = new Set();
+  protected override populateContainer(): void {
+    const dirtySpots = [
+      new DirtySpot(this.app, this),
+      new DirtySpot(this.app, this),
+      new DirtySpot(this.app, this),
+    ];
+    dirtySpots.forEach((spot) => {
+      this.remainingDirtySpots.add(spot);
+      this.container.addChild(spot.getObject());
+    });
+
+    this.container.eventMode = "static";
+    this.container.hitArea = this.app.screen;
+    const updateSpotsPointerDown = (isDown: boolean) => {

```

Comment:
Is there a reason we don't just have each dirty spot call delegate.isPointerDown and then just have a function on ScrubMinigame that returns of the pointer is currently down?

Reply:
I guess we could. But it'll be called for every single mousemove event (of which there'll be a lot)

Reply:
It'll be a boolean property lookup we'd set it to true and false based on the pointer down and up events. Basically identical to the current approach just a bit cleaner IMO.

But not fussed either way.

Reply:
Sure, updated in #28